Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -648,15 +655,19 @@ public class DependencyResolver<
         // Attempt to select each valid version in the preferred order.
         //
         // FIXME: We must detect recursion here.
-        return try AnySequence(validVersions(container).lazy.flatMap{ version -> AnySequence<AssignmentSet> in
+        return AnySequence(validVersions(container).lazy.flatMap{ version -> AnySequence<AssignmentSet> in
                 // Create an assignment for this container and version.
                 var assignment = AssignmentSet()
                 assignment[container] = .version(version)
 
+                // FIXME: Making these methods throwing will kill the lazy behavior.
+                guard let constraints = self.safely({ try container.getDependencies(at: version) }) else {
```

Comment:
This is possible to do for `getDependencies(at:)` and return an empty array but not for `getContainer(for:)` because we can't create dummy container in the resolver due to Container being a protocol